tests/apitests/robot/cmd.c

```diff
@@ -523,14 +523,29 @@ static void killnode(TestContext *context, int argc, char *argv[])
 static void restartnode(TestContext *context, int argc, char *argv[])
 {
     CarrierContextExtra *extra = context->carrier->extra;
+    CarrierContext *wctx = context->carrier;
     struct timeval timeout_interval;
     struct timeval now;
     extern void *carrier_run_entry(void *);
 
-    CHK_ARGS(argc == 2 || argc == 3);
+    CHK_ARGS(argc == 1 || argc == 2 || argc == 3);
 
     vlogI("Robot will be reborn.");
 
+    if (argc == 1) {
+        extra->restarting = true;
+
+        pthread_create(&extra->tid, 0, &carrier_run_entry, NULL);
+        // wait until carrier ready
+        cond_wait(wctx->cond);
+        // wait until joining persistent group
+        if (extra->groupid[0])
+            cond_wait(wctx->group_cond);
```

Comment:
where would invoke this condition (wctx->group_cond) ?

Reply:
Peer rejoining the group would have on_peer_list_changed callback invoked. This waits until the event taking place.